mlflow/utils/rest_utils.py

```diff
@@ -323,13 +323,10 @@ def cloud_storage_http_request(
     """
     if method.lower() not in ("put", "get", "patch", "delete"):
         raise ValueError("Illegal http method: " + method)
-    try:
-        with _get_http_response_with_retries(
-            method, url, max_retries, backoff_factor, retry_codes, timeout=timeout, **kwargs
-        ) as response:
-            yield response
-    except Exception as e:
-        raise MlflowException("API request failed with exception %s" % e)
```

Comment:
This prevents the following code from working as desired:

```python
try:
    with cloud_storage_http_request(...) as resp:
        resp.raise_for_status()
except requests.HTTPError as e:
    if should_retry(e):
        ...
```

Reply:
Oh man, great catch! Can we add a test case?

Reply:
We already have tests for the retry:

https://github.com/mlflow/mlflow/blob/8663d1ff718a2d88d448c997460a410eb4a516ee/tests/store/artifact/test_databricks_artifact_repo.py#L1271

but the root cause is mocking `mlflow.utils.rest_utils.cloud_storage_http_request`. In tests/store/artifact/test_databricks_artifact_repo.py, we have the following code to mock:

```python
with mock.patch("tests/store/artifact/test_databricks_artifact_repo.py", return_value=response):
    
```

this is equivalent to pretending:

```python
def cloud_storage_http_request(...):
    # Obiously not what cloud_storage_http_request actually does
    return response
```

and has been hiding the try-catch trap.